src/TodoContext.tsx

```diff
@@ -0,0 +1,317 @@
+  error: '',
+  existingCompleted: false,
+  nonCompletedTodos: 0,
+  disableInput: false,
+  isLoading: [],
+  tempTodo: null,
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  titleField: null!,
+  isChosenToRename: 0,
+  editingTodo: '',
+  handleEditing: 0,
+  setEditingTodo: () => {},
+  setIsLoading: () => {},
+  setError: () => {},
+  setFilter: () => {},
+  handleSubmit: () => {},
+  setPostTodo: () => {},
+  setFilteredTodos: () => {},
+  setDisableInput: () => {},
+  handleDelete: () => {},
+  handleCompletedDelete: () => {},
+  makeTodoCompleted: () => {},
+  setIsChosenToRename: () => {},
+  makeTodoChange: () => {},
+  setHandleEditing: () => {},
+});
+
+const USER_ID = 104;
+
+export const TodoProvider: React.FC<TodoContextProps> = ({ children }) => {
+  const [postTodo, setPostTodo] = useState('');
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filter, setFilter] = useState('all');
+  const [error, setError] = useState('');
+  const [disableInput, setDisableInput] = useState(false);
+  const [isLoading, setIsLoading] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  // const [doubleClickCounter, setDoubleClickCounter] = useState(0);
+  const [isChosenToRename, setIsChosenToRename] = useState(0);
+  const [editingTodo, setEditingTodo] = useState('');
+  const [handleEditing, setHandleEditing] = useState(0);
+  const [timeToFocus, setTimeToFocus] = useState(0);
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const titleField = useRef<HTMLInputElement>(null!);
+
+  useEffect(() => {
+    if (titleField?.current) {
+      titleField.current.focus();
+    }
+  }, [timeToFocus]);
+
+  const existingCompleted = todos.some((todo) => {
+    return todo.completed;
+  });
+
+  const nonCompletedTodos = todos.reduce((counter, todo) => {
+    if (!todo.completed) {
+      return counter + 1;
+    }
+
+    return counter;
+  }, 0);
+
+  useEffect(() => {
+    api.getTodos(USER_ID)
+      .then((apiTodos) => {
+        setTodos(apiTodos);
+        setFilteredTodos(apiTodos);
+        setTimeToFocus(current => current + 1);
+      })
+      .catch(() => {
+        setError('Unable to load todos');
+        setTimeToFocus(current => current + 1);
+      });
+  }, []);
+
+  useEffect(() => {
+    const timer = setTimeout(() => {
+      setError('');
+    }, 3000);
+
+    return () => {
+      clearTimeout(timer);
+    };
+  }, [error]);
+
+  useEffect(() => {
+    const filtereDtodo = () => {
+      switch (filter) {
+        case 'active':
+          return todos.filter((todo) => {
+            return todo.completed === false;
+          });
+        case 'completed':
+          return todos.filter((todo) => {
+            return todo.completed === true;
+          });
+        case 'all':
+          return todos;
+        default:
+          return todos;
+      }
```

Comment:
move this function to `utils`

Reply:
For code splitting and best code readability. 
General components/functions can be clearly.
We know this func does filtred todo ( because we have the name of this helprs/utils function), but we don't need a large part of the code in this main place.
If we need to check filtration we go to helprs/utils and check this part of the code, but inside the upperscope, we need fast orientation and less coun of code lines.